hyperbolic-rs/preprocess.py

```diff
@@ -104,63 +124,120 @@ def create_splits(samples, do_random=False, seed=42):
     print(f"Loading data from {item_item_file_path}")
     with tf.io.gfile.GFile(str(item_item_file_path), 'rb') as f:
         data = pickle.load(f)
     return data["item_item_distances"]
 
 
-def build_distance_matrix(item_item_distances_dict, iid2id, do_sparse=False, min_distance=0.1):
-    """
-    Build a distance matrix according to the graph distance between the items, stored in the item_item_distances_dict
-    The order of the matrix is given by the ids in iid2id.
-    This is, the distance between the item with numerical indexes i and j is in the position distance_matrix[i, j]
-
-    The distance to unconnected nodes or the distance from a node to itself is 0 if the matrix is sparse, or -1
-    if the matrix is dense.
-
-    :param item_item_distances_dict: dictionary that has the precomputed distances between pairs of items,
-    if there is a path between them in the semantic graph.
-    :param iid2id: mapping of item id (unique alpha numeric value that identifies the item) and
-    item index (0, 1, 2, ..)
-    :param do_sparse: if True, the distance matrix is returned as a sparse matrix, else as a numpy array
-    :param min_distance: minimum distance in the distance matrix. Values below will be clamped to min_distance.
+def build_item_item_triplets(item_item_distances_dict, iid2id, top_k):
     """
-    if do_sparse:
-        distance_matrix = np.zeros((len(iid2id), len(iid2id)))
-    else:
-        distance_matrix = np.ones((len(iid2id), len(iid2id))) * -1
+    Builds item item triples from the item-item distances
 
-    for src_iid, src_index in iid2id.items():
-        if src_iid not in item_item_distances_dict:
+    :param item_item_distances_dict: dict of src_iid: [(dst_iid, distance)]
+    :param iid2id: dict of item ids
+    :param top_k: adds top_k items per item at most
+    :return:
+    """
+    triplets = set()
+    for src_iid, dists in tqdm(item_item_distances_dict.items(), desc="item_item_triplets"):
+        if src_iid not in iid2id:
             continue
-        src_dist = item_item_distances_dict[src_iid]
-        for dst_iid, distance in src_dist.items():
-            if src_iid != dst_iid and dst_iid in iid2id:
-                dst_index = iid2id[dst_iid]
-                distance_matrix[src_index, dst_index] = max(distance, min_distance)
-    if do_sparse:
-        return sparse.csr_matrix(distance_matrix)
-    return distance_matrix
+        src_id = iid2id[src_iid]
+        sorted_dists = sorted(dists, key=lambda t: t[1])
+        added = 0
+        for dst_iid, cos_dist in sorted_dists:
+            if dst_iid not in iid2id or cos_dist > 0.3:
+                continue
+            dst_id = iid2id[dst_iid]
+            if cos_dist <= 0.1:
+                triplets.add((src_id, Relations.SEM_HIGH_SIM.value, dst_id))
+            elif 0.2 >= cos_dist > 0.1:
+                triplets.add((src_id, Relations.SEM_MEDIUM_SIM.value, dst_id))
+            else:   # 0.3 >= cos_dist > 0.2
```

Comment:
This definition / range is well hidden. Maybe put a comment in the enum?

Reply:
Done